server/src/main/scala/drt/server/feeds/lgw/ResponseToArrivals.scala

```diff
@@ -35,29 +35,31 @@ case class ResponseToArrivals(data: Array[Byte], locationOption: Option[String]
 
   def nodeToArrival: Node => (Arrival, Option[String]) = (n: Node) => {
 
+    val operator = (n \ "AirlineIATA") text
+    val actPax = parsePaxCount(n, "70A").filter(_!= 0).orElse(None)
+    val transPax = parsePaxCount(n, "TIP")
     val arrival = new Arrival(
-      Operator = (n \ "AirlineIATA") text,
+      Operator = if (operator.isEmpty) None else Some(operator) ,
       Status = parseStatus(n),
-      EstDT = parseDateTime(n, "TDN", "EST").getOrElse(""),
-      ActDT = parseDateTime(n, "TDN", "ACT").getOrElse(""),
-      EstChoxDT = parseDateTime(n, "ONB", "EST").getOrElse(""),
-      ActChoxDT = parseDateTime(n, "ONB", "ACT").getOrElse(""),
-      Gate = (n \\ "PassengerGate").headOption.map(n => n text).getOrElse(""),
-      Stand = (n \\ "ArrivalStand").headOption.map(n => n text).getOrElse(""),
-      MaxPax = (n \\ "SeatCapacity").headOption.map(n => (n text).toInt).getOrElse(0),
-      ActPax = parsePaxCount(n, "70A").getOrElse(0),
-      TranPax = parsePaxCount(n, "TIP").getOrElse(0),
-      RunwayID = parseRunwayId(n).getOrElse(""),
-      BaggageReclaimId = Try(n \\ "FIDSBagggeHallActive" text).getOrElse(""),
-      FlightID = 0,
+      Estimated = (((n \ "FlightLeg").head \ "LegData").head \\ "OperationTime").find(n => (n \ "@OperationQualifier" text).equals("TDN") && (n \ "@TimeType" text).equals("EST")).map(n => services.SDate.parseString(n text).millisSinceEpoch),
```

Comment:
This looks like repeated code, could it be put into a function, it will also make this more readable.

Reply:
parseDateTime looks like it already returns Option[String], could you not just update that to return Option[SDateLike]?